enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/BuiltInProceduresInteractionTestBase.java

```diff
@@ -638,17 +640,38 @@ public void shouldHaveSetTXMetaDataProcedure() throws Throwable
     }
 
     @Test
-    public void shouldFailOnTooLargeTXMetaData() throws Throwable
+    public void readUpdatedMetadataValue() throws Exception
     {
-        ArrayList<String> list = new ArrayList<>();
-        for ( int i = 0; i < 200; i++ )
+        GraphDatabaseFacade localGraph = neo.getLocalGraph();
+        try ( Transaction transaction = localGraph.beginTx() )
         {
-            list.add( format( "key%d: '0123456789'", i ) );
+            String testValue = "testValue";
+            String testKey = "test";
+            localGraph.execute( "CALL dbms.setTXMetaData", map( "data", map( testKey, testValue ) ) );
+            Map<String,Object> metadata =
+                    (Map<String,Object>) localGraph.execute( "CALL dbms.getTXMetaData " ).next().get( "metadata" );
```

Comment:
The `get("metadata")` seems unnecessary. The procedure already says `getTXMetaData` in the name. I think it would be more intuitive to have the return stream be a stream of tuples, with key:value columns.